Unable to compile with GCC 14

Garuda Linux (current) X86_64, GCC 14.1.1

Error as follows:
```
deps/libretro-common/encodings/encoding_utf.c: In function ‘utf8_to_local_string_alloc’:
deps/libretro-common/encodings/encoding_utf.c:359:14: error: implicit declaration of function ‘strdup’; did you mean ‘strldup’? [-Wimplicit-function-declaration]
359 |       return strdup(str);
|              ^~~~~~
|              strldup
deps/libretro-common/encodings/encoding_utf.c:359:14: error: returning ‘int’ from a function with return type ‘char *’ makes pointer from integer without a cast [-Wint-conversion]
359 |       return strdup(str);
|              ^~~~~~~~~~~
deps/libretro-common/encodings/encoding_utf.c: In function ‘local_to_utf8_string_alloc’:
deps/libretro-common/encodings/encoding_utf.c:374:14: error: returning ‘int’ from a function with return type ‘char *’ makes pointer from integer without a cast [-Wint-conversion]
374 |       return strdup(str);
|              ^~~~~~~~~~~
make: *** [Makefile:584: deps/libretro-common/encodings/encoding_utf.o] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -541,7 +541,7 @@ endif
 include Makefile.common
 
 ifeq (,$(findstring msvc,$(platform)))
-CFLAGS   += -std=c99
+CFLAGS   += -std=c99 -D_DEFAULT_SOURCE
 endif
 
 CFLAGS   += -Wall $(INCFLAGS) $(INCFLAGS_PLATFORM)

--- a/Makefile
+++ b/Makefile
@@ -541,7 +541,7 @@ endif
 include Makefile.common
 
 ifeq (,$(findstring msvc,$(platform)))
-CFLAGS   += -std=c99 -D_DEFAULT_SOURCE
+CFLAGS   += -std=gnu99
 endif
 
 CFLAGS   += -Wall $(INCFLAGS) $(INCFLAGS_PLATFORM)

